src/ImmersedBoundaries/mask_immersed_field.jl

```diff
@@ -1,10 +1,7 @@
 using KernelAbstractions: @kernel, @index
 using Statistics
 using Oceananigans.Architectures: architecture
-using Oceananigans.Fields: location, ZReducedField, Field
-
-instantiate(T::Type) = T()
-instantiate(t) = t
```

Comment:
It's ok to redefine `instantiate` here, since it makes the code easier to read (ie a case where "Write Everything Twice" (WET) is better than "Don't Repeat Yourself" (DRY). Just FYI for the future.

Reply:
just want to make this pass first 
https://buildkite.com/clima/oceananigans/builds/11941#01890352-3b10-49b6-98a3-0745c630a86c
then of course I agree on the comment

Reply:
Made a comment below to help